src/cashlib/cashlib.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) 2015-2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+/* clang-format off */
+// must be first for windows
+#include "compat.h"
+
+/* clang-format on */
+#include "base58.h"
+#include "random.h"
+#include "streams.h"
+#include "utilstrencodings.h"
+
+#include <openssl/rand.h>
+#include <string>
+#include <vector>
+
+static bool sigInited = false;
+
+// stop the logging
+int LogPrintStr(const std::string &str) { return str.size(); }
+namespace Logging
+{
+uint64_t categoriesEnabled = 0; // 64 bit log id mask.
+};
+
+// helper functions
+namespace
+{
+#if 0
+CKey LoadKey(uint256 val)
+{
+    CKey secret;
+    unsigned char *src = val.begin();
+    secret.Set(src, src+32, true);
+    return secret;
+}
+#endif
+
+CKey LoadKey(unsigned char *src)
+{
+    CKey secret;
+    secret.Set(src, src + 32, true);
+    return secret;
+}
+
+// From core_read.cpp #include "core_io.h"
+bool DecodeHexTx(CTransaction &tx, const std::string &strHexTx)
+{
+    if (!IsHex(strHexTx))
+        return false;
+
+    std::vector<unsigned char> txData(ParseHex(strHexTx));
+    CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
+    try
+    {
+        ssData >> tx;
+    }
+    catch (const std::exception &)
+    {
+        return false;
+    }
+
+    return true;
+}
+
+#if 0
+bool DecodeHexTx(CTransaction &tx, const char* cHexTx)
```

Comment:
remove disabled code

Reply:
The APIs provided in this library are useful which is why I issued the PR.  But are there other useful APIs?  While I was experimenting with APIs I wrote this code, and I'm not convinced that it won't be useful, although it isn't used now.

"cashlib" is not done.  New APIs will be useful, and other languages may need different formulations of the same API.  For example, passing data as hex strings rather then binary.  

But if we wait to PR until it is "done", it will be too large, and require constant merging.  I would like to merge cashlib in now as "experimental" (APIs may change), and continue to work on it with a series of smaller PRs.